generator/java/lib/Messages/MAVLinkPayload.java

```diff
@@ -51,85 +57,192 @@ public void resetIndex() {
     }
 
     public byte getByte() {
+
+        if (allZeros) return 0;
+
         byte result = 0;
-        result |= (payload.get(index + 0) & 0xFF);
-        index += 1;
+        try {
+            result |= (payload.get(index + 0) & 0xFF);
+            index += 1;
+        }
+        catch (final Exception ex)
+        {
+            allZeros = true;
+        }
         return result;
     }
 
     public short getUnsignedByte(){
+
+        if (allZeros) return 0;
+
         short result = 0;
-        result |= payload.get(index + 0) & 0xFF;
-        index+= 1;
-        return result; 
+        try {
+            result |= payload.get(index + 0) & 0xFF;
+            index+= 1;
+
+        }
+        catch (final Exception ex)
```

Comment:
Last I profiled this, this is a very expensive path to include. (It's been awhile since I've profiled it admittedly). Any time you hit a 0 trimmed packet you will be allocating a new exception object, and have to enter the handler, which ends up being slow and memory intensive.

It's usually easier to just either check that the index is in range first, or always ensure that the buffer has the expected minimum number of bytes allocated and filled out.

Reply:
I agree. You don't want to be throwing constant exceptions like this

Reply:
true.
@WickedShell  this is a valid concern.
guys, please check this [update](https://github.com/ArduPilot/pymavlink/pull/317/commits/6fd6ff15eef0fcf84440ab025cabba80ad5797bc)

I added:
final byte[] array = payload.array();
            final int len = array.length;
            if (index + LENGTH > len)
            {
                allZeros = true;
                return result;
            }

where LENGTH = 1,2,4.